src/managers/cluster.py

```diff
@@ -85,3 +88,67 @@ def update_credentials(self, username: str, password: str) -> None:
             client.update_password(username=username, new_password=password)
         except EtcdUserManagementError:
             raise
+
+    def add_member(self, unit_name: str) -> None:
+        """Add a new member to the etcd cluster."""
+        # retrieve the member information for the newly joined unit from the set of EtcdServers
+        member_name = ""
+        ip = ""
+        peer_url = ""
+
+        for server in self.state.servers:
+            if server.unit_name == unit_name:
+                member_name = server.member_name
+                ip = server.ip
+                peer_url = server.peer_url
+                break
+
+        # we need to make sure all required information are available before adding the member
+        if member_name and ip and peer_url:
+            try:
+                client = EtcdClient(
+                    username=self.admin_user,
+                    password=self.admin_password,
+                    client_url=self.state.unit_server.client_url,
+                )
+                cluster_members, member_id = client.add_member_as_learner(member_name, peer_url)
+                self.state.cluster.update(
+                    {"cluster_members": cluster_members, "learning_member": member_id}
+                )
+                logger.info(f"Added unit {unit_name} as new cluster member {member_id}.")
+            except EtcdClusterManagementError:
+                raise
+        else:
+            raise KeyError(f"Peer relation data for unit {unit_name} not found.")
+
+    def start_member(self) -> None:
+        """Start a cluster member and update its status."""
+        self.workload.start()
+        # this triggers a relation_changed event which the leader will use to promote
+        # a learner-member to fully-voting member
+        self.state.unit_server.update({"state": "started"})
+        if not self.state.cluster.cluster_state:
+            # mark the cluster as initialized
+            self.state.cluster.update(
+                {
+                    "cluster_state": "existing",
```

Comment:
Could you extract these state constants in a string enum? future typos can lead to hard to debug issues.